src/scripts/generate-binary-ops.py

```diff
@@ -0,0 +1,286 @@
+            op2 = _constants.min / op1
+        elif sum > _constants.max:
+            op2 = _constants.max / op1
+        op2 = int(op2) # Rounding because it might not be an integer
+        sum = op1 * op2
+        return {'oper': '*', 'op2': op2, 'sum': sum}
+
+    # Division (floor division)
+    def div(self, op1, op2):
+        if op2 == 0: # To avoid division by zero
+            if (random.randint(0,1)):
+                op2 = random.randint(1, _constants.max)
+            else:
+                op2 = random.randint(_constants.min, -1)
+        sum = op1 / op2
+        sum = int(sum) # Rounding because it might not be an integer
+        return {'oper': '/', 'op2': op2, 'sum': sum}
+
+    # Modulus
+    # Difference between % operator in python and js
+    # in python: -xxx % yy => yy - (abs(xxx) % yy)
+    # in js: -xxx % yy => - abs(xxx) % yy
+    def mod(self, op1, op2):
+        if op2 == 0:
+            if (random.randint(0,1)):
+                op2 = random.randint(1, _constants.max)
+            else:
+                op2 = random.randint(_constants.min, -1)
+        if (((op1 < 0 and op2 > 0) or (op1 > 0 and op2 < 0)) and (op1 % op2 != 0)):
+            sum = ((op1 % op2) - op2)
+        else:
+            sum = (op1 % op2)
+        return {'oper': '%', 'op2': op2, 'sum': sum}
+
+    # Bitwise operations
+
+    # Bitshifting to left
+    def bitshiftleft(self, op1, op2):
+        op2 = random.randint(_constants.bitmin, _constants.bitmax)
+        if op1 >= _constants.bitmax or op1 <= _constants.bitmin:
+            op2 = 0
+        sum = int(hex((op1 << ((op2 % (_constants.bitmax_exposant + 1)) & 0x1f))), 16)
+        if sum > _constants.bitmax or sum < _constants.bitmin: # Operand 2 must depend on the length of operand 1
+            if op1 != 0:
+                op2 = (_constants.bitmax_exposant - 1) - int.bit_length(op1)
+                if op2 < 0: # Operand 2 also cant go below zero
+                    op2 = 0
+        sum = int(hex((op1 << ((op2 % (_constants.bitmax_exposant + 1)) & 0x1f))), 16)
+        return {'oper': '<<', 'op2': op2, 'sum': sum}
+
+    # Bitshifting to right
+    # Python only can handle shiftcount between 0-31, but smaller and bigger numbers
+    # are also can be simulated with this method
+    def bitshiftright(self, op1, op2):
+        op2 = random.randint(_constants.bitmin, _constants.bitmax)
+        sum = int(hex((op1 >> ((op2 % (_constants.bitmax_exposant + 1)) & 0x1f))), 16)
+        return {'oper': '>>', 'op2': op2, 'sum': sum}
+
+    # Unsigned right bitshifting
+    def bitshiftrightzero(self, op1, op2):
+        op2 = random.randint(_constants.bitmin, _constants.bitmax)
+        if (op1 >= 0): # For positive operand 1 it works the same as bitshiftright
+            sum = int(hex((op1 >> ((op2 % (_constants.bitmax_exposant + 1)) & 0x1f))), 16)
+        else: # For negative operand 1 it needs conversion to unsigned
```

Comment:
Still has issue related to the standard. See the Runtime Semantics: Evaluation part at https://www.ecma-international.org/ecma-262/6.0/#sec-unsigned-right-shift-operator . The op1 and op2 have to be unsigned int32.